components/infobox/extensions/commons/years_active_base.lua

```diff
@@ -0,0 +1,157 @@
+		player = player:gsub('_', ' ')
+	end
+
+	-- since TeamCards on some wikis store players with underscores and some with spaces
+	-- we need to check for both options
+	local playerAsPageName = player:gsub(' ', '_')
+
+	local prefix = args.prefix or 'p'
+
+	local playerPositionLimit = tonumber(args.playerPositionLimit) or ActiveYears.defaultNumberOfStoredPlayersPerMatch
+	if playerPositionLimit <=0 then
+		error('"playerPositionLimit" has to be >= 1')
+	end
+
+	local conditions = '([[participant::' .. player .. ']] OR [[participant::' .. playerAsPageName .. ']]'
+	for playerIndex = 1, playerPositionLimit do
+		conditions = conditions .. ' OR [[players_' .. prefix .. playerIndex .. '::' .. player .. ']]'
+		conditions = conditions .. ' OR [[players_' .. prefix .. playerIndex .. '::' .. playerAsPageName .. ']]'
+	end
+	conditions = conditions .. ')'
+
+	conditions = conditions .. ' AND [[date::!' .. _DEFAULT_DATE .. ']] AND (' ..
+		'[[date_year::>' .. ActiveYears.startYear .. ']] OR ' ..
+		'[[date_year::' .. ActiveYears.startYear .. ']])'
+
+	if String.isNotEmpty(args.mode) then
+		conditions = conditions .. ' AND [[mode::' .. args.mode .. ']]'
+	end
+
+	conditions = conditions .. ActiveYears.additionalConditions
+
+	return ActiveYears._calculate(conditions)
+end
+
+function ActiveYears._calculate(conditions)
+	local years = {}
+	local offset = 0
+	local count = _MAX_QUERY_LIMIT
+
+	while count == _MAX_QUERY_LIMIT do
+		local lpdbQueryData = mw.ext.LiquipediaDB.lpdb('placement', {
+			order = 'date asc',
+			conditions = conditions,
+			query = 'date',
+			limit = _MAX_QUERY_LIMIT,
+			offset = offset
+		})
+
+		if offset == 0 and #lpdbQueryData == 0 then
+			return 'Player has no results.'
+		end
+
+		-- Find all years for which the player has at least one placement
+		for _, item in ipairs(lpdbQueryData) do
+			local year = tonumber(string.sub(item.date, 1, 4))
+			years[year] = 0
+		end
+
+		count = #lpdbQueryData
+		offset = offset + _MAX_QUERY_LIMIT
+	end
+
+	-- Sort years chronologically
+	local sortedYears = {}
+	for year in pairs(years) do
+		table.insert(sortedYears, year)
+	end
+	table.sort(sortedYears)
+
+	-- Determine activity ranges by grouping consecutive years
+	local startYear
+	local endYear
+	local yearRanges = {}
+	for index, year in ipairs(sortedYears) do
+		if index == 1 then
+			startYear = year
+			endYear = year
+		else
+			if year - endYear == 1 then
+				endYear = year
+			else
+				if startYear == endYear then
+					table.insert(yearRanges, tostring(startYear))
+				else
+					table.insert(yearRanges, tostring(startYear) .. ' - ' .. tostring(endYear))
+				end
+				startYear = year
+				endYear = year
+			end
+		end
+	end
+	if endYear == _CURRENT_YEAR then
+		table.insert(yearRanges, tostring(startYear) .. ' - ' .. "'''Present'''")
+	else
+		if startYear == endYear then
+			table.insert(yearRanges, tostring(startYear))
+		else
+			table.insert(yearRanges, tostring(startYear) .. ' - ' .. tostring(endYear))
+		end
+	end
```

Comment:
Lines 123-127 and 136-140 are carbon copies of eachother. Feels like the code overall should be possible to write cleaner as well.